hera_cal/redcal.py

```diff
@@ -1007,17 +1015,26 @@ def redcal_partial_io_iteration(hd, nInt_to_load=8, pol_mode='2pol', ex_ants=[],
     if verbose and np.any(solar_flagged):
         print(len(hd.times[solar_flagged]), 'integrations flagged due to sun above', solar_horizon, 'degrees.')
     
-    # loop over data, performing partial loading 
+    # loop over polarizations and times, performing partial loading if desired
     for pols in pol_load_list:
         if verbose:
             print('Now calibrating', pols, 'polarization(s)...')
         reds = filter_reds(all_reds, ex_ants=ex_ants, pols=pols)
-        for tinds in np.split(np.arange(nTimes)[~solar_flagged], 
-                              np.arange(nInt_to_load, len(hd.times[~solar_flagged]), nInt_to_load)):
+        if nInt_to_load > 0:  # split up the integrations to load nInt_to_load at a time
+            tind_groups = np.split(np.arange(nTimes)[~solar_flagged], 
```

Comment:
Can split handle a case where nInt_to_load is the whole file, in order to avoid if/else?